kombu/transport/SQS.py

```diff
@@ -156,6 +156,48 @@ class UndefinedQueueException(Exception):
     """Predefined queues are being used and an undefined queue was used."""
 
 
+class QoS(virtual.QoS):
+    def reject(self, delivery_tag, requeue=False):
+        super().reject(delivery_tag, requeue=requeue)
+        queue_name, message, backoff_tasks, backoff_policy = self._extract_backoff_policy_configuration_and_message(delivery_tag)
+        if queue_name and backoff_tasks and backoff_policy:
+            self.apply_backoff_policy(queue_name, delivery_tag, backoff_policy, backoff_tasks)
+
+    def _extract_backoff_policy_configuration_and_message(self, delivery_tag):
+        try:
+            message = self._delivered[delivery_tag]
+            routing_key = message.delivery_info['routing_key']
+        except KeyError:
+            return None, None, None
+        if not routing_key or not message:
+            return None, None, None
+        queue_config = self.channel.predefined_queues.get(routing_key, {})
+        backoff_tasks = queue_config.get('backoff_tasks')
+        backoff_policy = queue_config.get('backoff_policy')
+        return routing_key, message, backoff_tasks, backoff_policy
+
+    def apply_backoff_policy(self, queue_name, delivery_tag, backoff_policy, backoff_tasks):
+        queue_url = self.channel._queue_cache[queue_name]
+        task_name, number_of_retries = self.extract_task_name_and_number_of_retries(delivery_tag)
+        if not task_name or not number_of_retries:
+            return None
+        policy_value = backoff_policy.get(number_of_retries)
```

Comment:
This is reasonable, but would you say that adding a 'max_retries' (or a better name) key is a valid improvement?
If so, would you mind issuing another PR?

Reply:
I don't mind creating a new PR, but bear in mind that the default visibility timeout is configured in SQS so practically we are overriding the queue configuration

Reply:
In that case, we should note that in the documentation.

Reply:
@thedrow @auvipy created new PR with doc change. feel free suggesting different wording https://github.com/celery/kombu/pull/1304

Reply:
@thedrow same here https://github.com/celery/celery/pull/6653